src/subscription_manager/utils.py

```diff
@@ -263,7 +263,6 @@ def get_server_versions(cp, exception_on_timeout=False):
             # instead of waiting for all the calls to timeout
             if exception_on_timeout:
                 log.error("Timeout error while checking server version")
-                raise
```

Comment:
We should leave this here otherwise the "exception_on_timeout" value is meaningless.

I suggest we look at where this method is used and wrap its usage (with the exception_on_timeout value passed in as true) in a try catch